modules/optimization/src/vectorpostprocessors/ElementOptimizationReactionFunctionInnerProduct.C

```diff
@@ -0,0 +1,37 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElementOptimizationReactionFunctionInnerProduct.h"
+
+registerMooseObject("OptimizationApp", ElementOptimizationReactionFunctionInnerProduct);
+
+InputParameters
+ElementOptimizationReactionFunctionInnerProduct::validParams()
+{
+  InputParameters params = ElementOptimizationFunctionInnerProduct::validParams();
+  params.addClassDescription(
+      "Compute the gradient for material inversion by taking the inner product of the "
```

Comment:
This should be reworded for this type of material inversion problem where its only a function of the time derivative and not the spatial derivative.